include/boost/math/differentiation/autodiff.hpp

```diff
@@ -1239,7 +1268,8 @@ fvar<RealType,Order> frexp(const fvar<RealType,Order>& cr, int* exp)
 template<typename RealType, size_t Order>
 fvar<RealType,Order> ldexp(const fvar<RealType,Order>& cr, int exp)
 {
-    return cr * std::exp2(exp);
+    // argument to std::exp2 must be casted to RealType, otherwise std::exp2 returns double (always)
+    return cr * std::exp2(static_cast<RealType>(exp));
```

Comment:
I don't think this will compile when `RealType` is an AD type. E.g. `cr` is a `autodiff_fvar<double,3,4>`.

Is there somewhere in a special_function where this is required?

Reply:
`tangent_scale_factor` via `tangent_t2n` for long double